go-controller/pkg/util/net_linux.go

```diff
@@ -681,3 +695,52 @@ func GetIPFamily(v6 bool) int {
 func (defaultNetLinkOps) LinkSetVfHardwareAddr(pfLink netlink.Link, vfIndex int, hwaddr net.HardwareAddr) error {
 	return netlink.LinkSetVfHardwareAddr(pfLink, vfIndex, hwaddr)
 }
+
+func findIPInNetworkFromIface(dstIP net.IP, iface net.Interface) (net.IP, error) {
```

Comment:
having `dstIP` as the parameter name is confusing ? I hope it's about finding an interface containing that ip address. can you just change the name to `ipAddr` or something like that ? it may also return `bool` instead of `net.IP`

Reply:
This is a copy of https://github.com/j-keck/arping/blob/c50f4319d307d95d1e2b344c4504158705e919de/netutils.go#L9

But I can adapt it a little to fit better at our codebase.